runtime/tests/org.eclipse.core.tests.runtime/src/org/eclipse/core/tests/runtime/jobs/JobTest.java

```diff
@@ -1002,18 +991,20 @@ public void testJoinWithCancelingMonitor() {
 	}
 
 	public void testJoinInterruptNonUIThread() throws InterruptedException {
+		TestBarrier2 barrier = new TestBarrier2();
 		final Job job = new TestJob("job", 50000, 100);
 		Thread t = new Thread(() -> {
 			job.schedule();
 			try {
+				barrier.setStatus(Job.RUNNING);
```

Comment:
(i have not looked at the whole commit)
this change does totally change the test and probably render it meaningless: scheduling a job does not mean the job is already started.

Reply:
The comment is correct w.r.t. naming, i.e., the barrier actually ensures the job is scheduled and not necessarily running. Thanks for pointing that out.
However, being scheduled in enough to keep the semantics of the test. The test is about interrupting a blocked join operation, which is still given when the job is scheduled and not necessarily running. Thus the tests remains meaningful and in particular validates the same functionality as before, but now with deterministic behavior.

I will update the PR with proper naming for the barrier.

Reply:
I have updated the commit with proper naming for the barrier status and the according comment